src/Assignments/SimpleProblemPage.tsx

```diff
@@ -6,98 +6,305 @@ import ProblemIframe from './ProblemIframe';
+                });
+            }
+
+            setModalLoading(false);
+        } catch (e) {
+            setError(e.message);
+            setModalLoading(false);
+        }
+    };
+
+    const loadNewVersion = async (topic: TopicObject) => {
+        setModalLoading(true);
+        if (_.isNil(topic)) {
+            console.error('This should not happen - no topic loaded');
+        } else {
+            try {
+                const res = await generateNewVersion({ topicId: topic.id });
+                const version = res.data.data;
+                if (_.isNil(topic.studentTopicAssessmentInfo) === false && topic.studentTopicAssessmentInfo.length > 0) {
+                    topic.studentTopicAssessmentInfo.push(version);
+                } else {
+                    topic.studentTopicAssessmentInfo = [version];
+                }
+                setTopic(topic);
+                if (_.isNil(version.id)) {
+                    console.error('we tried to generate a new version, but failed.');
+                } else {
+                    setVersionId(version.id);
+                    fetchProblems(topic.id);
+                }
+                setModalLoading(false);
+            } catch (e) {
+                setError(e.message);
+                setModalLoading(false);
+            }
+        }
+    };
+
+    const endCurrentVersion = async (versionId: number) => {
+        // const result = await endVersion({topicId, versionId});
+        // if zero attemptsRemaining, we don't need to tell the backend to close
+        if (!_.isNil(attemptsRemaining) && attemptsRemaining > 0) {
+            if (_.isNil(topic)) {
+                console.error('This should not happen - no topic loaded');
+            } else {
+                setModalLoading(true);
+                try {
+                    console.log('drew: ', attemptsRemaining, ' attempts remaining and ', versionsRemaining, ' versions remaining.');
+                    const res = await endVersion({ versionId });
+                    console.log('drew: ', res);
+                    fetchProblems(topic.id); // reload the problems in case they are supposed to be hidden after close
+                } catch (e) {
+                    setError(e.message);
+                    console.error(e);
+                    setConfirmationParameters(DEFAULT_CONFIRMATION_PARAMETERS);
+                    setModalLoading(false);
+                }
+                setModalLoading(false);
+            }
+        } else {
+            // student had no attempts remaining, so the backend already knows the version is closed
+            setConfirmationParameters(DEFAULT_CONFIRMATION_PARAMETERS);
+            setModalLoading(false);
+        }
+    };
+
+    const generateScoreTable = (data: any) => {
+        const { problemScores, bestVersionScore, bestOverallVersion } = data;
+        return (
+            <div className="d-flex flex-column">
```

Comment:
Absolutely fine for now, however, I think this is one of those cases where we have the code styled to the developer, we've been using a lot of material UI grids, I wonder if this should be refactored to be consistent in the future

@PanopticaRising thoughts?

Reply:
I prefer that we're consistent as well, eventually. We'll do a design review and try to land on something consistent.